RecoLocalCalo/Configuration/python/RecoLocalCalo_cff.py

```diff
@@ -32,37 +32,29 @@
 from RecoLocalCalo.Configuration.hcalLocalRecoNZS_cff import *
 calolocalrecoNZS = cms.Sequence(ecalLocalRecoSequence+hcalLocalRecoSequence+hcalLocalRecoSequenceNZS) 
 
-def _modifyRecoLocalCaloConfigurationReconstructorsForPhase2CommonHBHE( obj ):
-    obj.digiLabel = cms.InputTag('simHcalDigis','HBHEUpgradeDigiCollection')
-
-def _modifyRecoLocalCaloConfigurationReconstructorsForPhase2CommonHO( obj ):
-    obj.digiLabel = cms.InputTag('simHcalDigis')
-
-def _modifyRecoLocalCaloConfigurationReconstructorsForPhase2CommonHF( obj ):
-    obj.digiLabel = cms.InputTag('simHcalDigis','HFUpgradeDigiCollection')
-
-def _modifyRecoLocalCaloConfigurationReconstructorsForPhase2CommonZDC( obj ):
-    obj.digiLabel = cms.InputTag('simHcalUnsuppressedDigis')
-
 def _modifyRecoLocalCaloConfigurationProcessForPhase2Common( theProcess ):
-    theProcess.load("RecoLocalCalo.HcalRecProducers.HBHEUpgradeReconstructor_cfi")
-    theProcess.load("RecoLocalCalo.HcalRecProducers.HFUpgradeReconstructor_cfi")
+    from RecoLocalCalo.HcalRecProducers.HBHEUpgradeReconstructor_cfi import hbheUpgradeReco as _hbheUpgradeReco
+    from RecoLocalCalo.HcalRecProducers.HFUpgradeReconstructor_cfi import hfUpgradeReco as _hfUpgradeReco
+    theProcess.hbheUpgradeReco = _hbheUpgradeReco
+    theProcess.hfUpgradeReco = _hfUpgradeReco
     theProcess.hcalLocalRecoSequence.replace(theProcess.hfreco,theProcess.hfUpgradeReco)
     theProcess.hcalLocalRecoSequence.remove(theProcess.hbhereco)
     theProcess.hcalLocalRecoSequence.replace(theProcess.hbheprereco,theProcess.hbheUpgradeReco)
```

Comment:
I don't remember what @Dr15Jones commented about these, but if `process.hbheUpgradeReco` and `process.hfUpgradeReco` are defined only here, the `ProcessModifier` (i.e. this function and the corresponding `makeProcessModifier()`) could be replaced with something along
```
from RecoLocalCalo.HcalRecProducers.HBHEUpgradeReconstructor_cfi import hbheUpgradeReco # or import *?
from RecoLocalCalo.HcalRecProducers.HFUpgradeReconstructor_cfi import hfUpgradeReco # or import *?

_hcalLocalRecoSequence_phase2_common = hcalLocalRecoSequence.copyAndExclude([hbhereco])
_hcalLocalRecoSequence_phase2_common.replace(hfreco, hfUpgradeReco)
_hcalLocalRecoSequence_phase2_common.replace(hbheprereco, hbheUpgradeReco)
eras.phase2_common.toReplaceWith(hcalLocalRecoSequence, _hcalLocalRecoSequence_phase2_common)
```

I also think the customization of `hcalLocalRecoSequence` would be safer and clearer to do in `hcalLocalReco_cff.py` where the sequence is defined.